cpp/src/robotick/framework/data/RemoteEngineConnection.cpp

```
@@ -0,0 +1,1088 @@
+				ROBOTICK_WARNING("Failed to set SO_REUSEADDR on socket");
+			}
+
+			// Disable Nagle's algorithm (batches up messages) — send every message immediately (essential for real-time control)
+			if (setsockopt(fd, IPPROTO_TCP, TCP_NODELAY, &opt, sizeof(opt)) < 0)
+			{
+				close(fd);
+				ROBOTICK_WARNING("Failed to set TCP_NODELAY on socket");
+			}
+
+			// Set non-blocking
+			int flags = fcntl(fd, F_GETFL, 0);
```

Clang (14.0.6)

[warning] variable 'flags' is not initialized

(cppcoreguidelines-init-variables)